qtop.py

```diff
@@ -535,76 +595,111 @@ def create_user_accounts_pool_mappings(account_jobs_table):
+    anywhere in the given matrix
     """
     # lines = []
-    for ind, k in enumerate(cpu_core_dict):
-        color_cpu_core_list = list(
-            line_with_separators(
-                cpu_core_dict['Cpu' + str(ind) + 'line'][print_start:print_end],
-                SEPARATOR,
-                options.WN_COLON
-            )
-        )
-        color_cpu_core_list = [colorize(elem, pattern_of_id[elem]) for elem in color_cpu_core_list if elem in pattern_of_id]
-        line = ''.join(color_cpu_core_list)
-        yield line + colorize('=Core' + str(ind), 'account_not_coloured')
+    for ind, k in enumerate(core_user_map):
+        cpu_core_line = core_user_map['Core' + str(ind) + 'line'][print_start:print_end]
+        if options.REM_EMPTY_CORELINES and '#' * (print_end - print_start) == cpu_core_line:
+            continue
+        cpu_core_line = insert_separators(cpu_core_line, SEPARATOR, options.WN_COLON)
+        cpu_core_line = ''.join([colorize(elem, pattern_of_id[elem]) for elem in cpu_core_line if elem in pattern_of_id])
+        yield cpu_core_line + colorize('=Core' + str(ind), 'account_not_coloured')
+
+
+def calc_core_userid_matrix(cluster_dict, id_of_username, job_ids, user_names):
+    _core_user_map = OrderedDict()
+    max_np_range = [str(x) for x in range(cluster_dict['max_np'])]
+    user_of_job_id = dict(izip(job_ids, user_names))
 
+    for core_nr in max_np_range:
+        _core_user_map['Core' + str(core_nr) + 'line'] = []  # Cpu0line, Cpu1line, Cpu2line, .. = '','','', ..
 
-def calc_cpu_lines(node_dict, id_of_username, job_ids, user_names):
-    _cpu_core_dict = {}
-    max_np_range = []
-    user_of_job_id = dict(izip(job_ids, user_names))
+    for _node in cluster_dict['wn_dict']:
+        state_np_corejob = cluster_dict['wn_dict'][_node]
+        _core_user_map, cluster_dict['wn_dict'][_node]['core_user_column'] = fill_node_cores_column(state_np_corejob, _core_user_map, id_of_username, max_np_range, user_of_job_id)
 
-    for core_nr in range(node_dict['max_np']):
-        _cpu_core_dict['Cpu' + str(core_nr) + 'line'] = ''  # Cpu0line, Cpu1line, Cpu2line, .. = '','','', ..
-        max_np_range.append(str(core_nr))
+    for coreline in _core_user_map:
+        _core_user_map[coreline] = ''.join(_core_user_map[coreline])
 
-    for _node in node_dict['wn_dict']:
-        state_np_corejob = node_dict['wn_dict'][_node]
-        _cpu_core_dict = fill_cpucore_columns(state_np_corejob, _cpu_core_dict, id_of_username, max_np_range, user_of_job_id)
+    return _core_user_map
 
-    return _cpu_core_dict
 
+def calc_attr():
+    pass
 
-def calculate_wn_occupancy(node_dict, user_names, job_states, job_ids):
+
+def get_yaml_key_part(major_key):
+    """
+    only return the list items of the yaml major_key if a yaml key subkey exists
+    (this signals a user-inserted value)
+    """
+    # e.g. major_key = 'wn_occupancy'
+    for part in config[major_key]:
+        part_name = [i for i in part][0]
```

Comment:
[0] implies something like a key? just confirm this is appropriate type of data structure